File: apps/cron/src/jobs/verifications.ts
import type { Database } from "@raven/db";
import { verifications } from "@raven/db";
import { lt } from "drizzle-orm";

export const cleanupExpiredVerifications = async (db: Database) => {
  const now = new Date();

  const deleted = await db
    .delete(verifications)
    .where(lt(verifications.expiresAt, now))
    .returning({ id: verifications.id });

  console.log(
    `Verification cleanup: deleted ${deleted.length} expired verifications`
  );
};

File: apps/cron-py/src/jobs/verifications.py
def cleanup_expired_verifications(conn) -> None:
    with conn.cursor() as cur:
        cur.execute("""
            DELETE FROM verifications
            WHERE "expiresAt" < NOW()
        """)
        count = cur.rowcount
        if count > 0:
            print(f"  Deleted {count} expired verification(s)")
